res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -355,40 +353,92 @@ ReloopBeatmix24.WheelTurn = function(channel, control, value, status, group) {
     }
 };
 
+// Led Feedback functions
+// ========================================================
+ReloopBeatmix24.AllJogLEDsToggle = function(deck, state, step) {
+    var step = typeof step !== 'undefined' ? step : 1; // default value
+    for (var j = 0x30; j <= 0x3F; j += step) {
+        midi.sendShortMsg(deck, j, state);
+    }
+};
+
 ReloopBeatmix24.deckloaded = function(value, group, control) {
     var i;
     switch (group.substr(1, 7)) {
         case "Channel":
             var channelRegEx = /\[Channel(\d+)\]/;
-            var channelChan = parseInt(channelRegEx.exec(group))[1];
+            var channelChan = parseInt(channelRegEx.exec(group)[1]);
             if (channelChan <= 4) {
                 midi.sendShortMsg(0x90 + channelChan, 0x50, value ? ON :
                     OFF);
             }
             break;
         case "Sampler":
             var samplerRegEx = /\[Sampler(\d+)\]/;
-            var samplerChan = parseInt(samplerRegEx.exec(group))[1];
-            if (samplerChan <= 4) { // We only handle 4 samplers (1 per pad)
+            var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+            if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
                 for (i = 0x91; i <= 0x94; i++) {
                     // PAD1 Mode A
                     midi.sendShortMsg(i, 0x08 - 1 + samplerChan, value ?
                         RED : OFF);
                     // SHIFT+PAD1 Mode A
                     midi.sendShortMsg(i, 0x48 - 1 + samplerChan, value ?
                         RED : OFF);
-                    // PAD5 Mode A+B
-                    midi.sendShortMsg(i, 0x14 - 1 + samplerChan, value ?
-                        RED : OFF);
-                    // SHIFT+PAD5 Mode A+B
-                    midi.sendShortMsg(i, 0x1C - 1 + samplerChan, value ?
-                        RED : OFF);
+                    if (samplerChan <= 4) { // Handle first 4 samplers in split mode
+                        // PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x14 - 1 + samplerChan, value ?
+                            RED : OFF);
+                        // SHIFT+PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x1C - 1 + samplerChan, value ?
+                            RED : OFF);
+                    }
                 }
             }
             break;
     }
 };
 
+ReloopBeatmix24.SamplerPlay = function(value, group, control) {
+    var samplerRegEx = /\[Sampler(\d+)\]/;
+    var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+    if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
```

Comment:
because sampler pads are only in "Mode B" and you may need, say, loop control on one channel and also need sampler at the same time. So I choose to map the 4 channel pads to the same samplers (this way, you can control 8 samplers).